tests/test_exceptions.py

```diff
@@ -16,66 +14,39 @@ def test_httpcore_all_exceptions_mapped() -> None:
     All exception classes exposed by HTTPCore are properly mapped to an HTTPX-specific
     exception class.
     """
-    not_mapped = [
-        value
-        for name, value in vars(httpcore).items()
+    expected_mapped_httpcore_exceptions = {
+        value.__name__
+        for _, value in vars(httpcore).items()
         if isinstance(value, type)
         and issubclass(value, Exception)
-        and value not in HTTPCORE_EXC_MAP
         and value is not httpcore.ConnectionNotAvailable
-    ]
+    }
 
-    if not_mapped:  # pragma: no cover
-        pytest.fail(f"Unmapped httpcore exceptions: {not_mapped}")
+    httpx_exceptions = {
+        value.__name__
+        for _, value in vars(httpx).items()
+        if isinstance(value, type) and issubclass(value, Exception)
+    }
 
+    unmapped_exceptions = expected_mapped_httpcore_exceptions - httpx_exceptions
 
-def test_httpcore_exception_mapping(server: "TestServer") -> None:
-    """
-    HTTPCore exception mapping works as expected.
-    """
-
-    def connect_failed(*args, **kwargs):
-        raise httpcore.ConnectError()
-
-    class TimeoutStream:
-        def __iter__(self):
-            raise httpcore.ReadTimeout()
-
-        def close(self):
-            pass
-
-    with mock.patch(
-        "httpcore.ConnectionPool.handle_request", side_effect=connect_failed
-    ):
-        with pytest.raises(httpx.ConnectError):
-            httpx.get(server.url)
+    if unmapped_exceptions:  # pragma: no cover
+        pytest.fail(f"Unmapped httpcore exceptions: {unmapped_exceptions}")
 
-    with mock.patch(
-        "httpcore.ConnectionPool.handle_request",
-        return_value=httpcore.Response(
-            200, headers=[], content=TimeoutStream(), extensions={}
-        ),
-    ):
-        with pytest.raises(httpx.ReadTimeout):
-            httpx.get(server.url)
 
-
-def test_httpx_exceptions_exposed() -> None:
```

Comment:
The purpose of this test was to verify that we don't leave any exception class defined in `httpx/_exceptions.py` out of `httpx/__init__.py`.

As such, it's impossible to perform without reaching to `httpx._exceptions`.

But I also thought it wasn't very useful after all, so I am dropping it in this PR.